src/ucp/core/ucp_worker.c

```diff
@@ -1209,21 +1209,26 @@ static void ucp_worker_iface_set_sys_device_distance(ucp_worker_iface_t *wiface)
     ucs_sys_dev_distance_t *distance = &wiface->distance;
     ucs_sys_device_t device          = UCS_SYS_DEVICE_ID_UNKNOWN;
     ucs_sys_device_t cmp_device      = UCS_SYS_DEVICE_ID_UNKNOWN;
-    ucp_rsc_index_t md_index, i;
+    ucp_rsc_index_t i;
+    char *md_name;
+    char buf[128];
 
     *distance = ucs_topo_default_distance;
 
     for (i = 0; i < context->num_tls; i++) {
-        md_index = context->tl_rscs[i].md_index;
-        if (strcmp(context->tl_mds[md_index].rsc.md_name,
-                   context->config.ext.select_distance_md)) {
+        md_name = context->tl_mds[context->tl_rscs[i].md_index].rsc.md_name;
+        if (strcmp(md_name, context->config.ext.select_distance_md)) {
             continue;
         }
 
         device     = ucp_worker_iface_get_sys_device(wiface);
         cmp_device = context->tl_rscs[i].tl_rsc.sys_device;
 
         ucs_topo_get_distance(device, cmp_device, distance);
+        ucs_trace("distance between iface[%p]/dev[%d] and md[%s]/dev[%d] is %s",
+                  wiface->iface, device, md_name, cmp_device,
```

Comment:
shall we print the resource name and index instead of iface ptr?

Reply:
Done